h2o-genmodel/src/main/java/hex/genmodel/easy/OneHotEncoderColumnMapper.java

```diff
@@ -18,11 +18,20 @@ public OneHotEncoderColumnMapper(GenModel m) {
     String[][] origDomainValues = _m.getOrigDomainValues();
     Map<String, Integer> columnMapping = new HashMap<>(origNames.length);
     int pos = 0;
+    // non-categorical
+    for (int i = 0; i < _m.getOrigNumCols(); i++) {
```

Comment:
We need to iterate twice, because the number of null `origDomainValues` is not known beforehand, correct ? Thus we can't do only one `for` cycle, writing categoricals with proper offset righ away ?

Reply:
That is correct, it could be simplified but this code runs just once on load. IMHO not an issue to do 2 for-loops.